File: app/src/main/java/me/pwcong/jpstart/component/fragment/GameFragment.java
package me.pwcong.jpstart.component.fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.CardView;
import android.view.View;

import butterknife.BindView;
import me.pwcong.jpstart.R;
import me.pwcong.jpstart.rxbus.RxBus;
import me.pwcong.jpstart.rxbus.event.EventContainer;
import me.pwcong.jpstart.rxbus.event.GameEvent;

/**
 * Created by Pwcong on 2016/10/24.
 */

public class GameFragment extends BaseFragment {

    @BindView(R.id.cv_puzzle)
    CardView mPuzzleCardView;
    @BindView(R.id.cv_supperzzle)
    CardView mSupperzzleCardView;

    @Override
    protected int getViewId() {
        return R.layout.fragment_game;
    }

    @Override
    protected void initVariable(@Nullable Bundle savedInstanceState) {

        mPuzzleCardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                RxBus.getDefault().post(new EventContainer(EventContainer.TYPE_GAME, new GameEvent(GameEvent.TYPE_PUZZLE)));
            }
        });

        mSupperzzleCardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                RxBus.getDefault().post(new EventContainer(EventContainer.TYPE_GAME, new GameEvent(GameEvent.TYPE_SUPPERZZLE)));
            }
        });

    }

    @Override
    protected void doAction() {

    }


}

File: app/src/main/java/com/github/pwcong/jpstart/view/fragment/GameFragment.kt
package com.github.pwcong.jpstart.ui.fragment


import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.cardview.widget.CardView
import com.github.pwcong.jpstart.R
import com.github.pwcong.jpstart.databinding.FragmentGameBinding
import com.github.pwcong.jpstart.rxbus.RxBus
import com.github.pwcong.jpstart.rxbus.event.EventContainer
import com.github.pwcong.jpstart.rxbus.event.GameEvent

class GameFragment : BaseFragment<FragmentGameBinding>() {

    private lateinit var mPuzzleCardView: CardView

    override fun initViewBinding(
        inflater: LayoutInflater,
        container: ViewGroup?
    ): FragmentGameBinding {
        return FragmentGameBinding.inflate(inflater, container, false)
    }

    override fun initVariable(savedInstanceState: Bundle?) {
        mPuzzleCardView = getViewBinding().root.findViewById(R.id.cv_puzzle)
        mPuzzleCardView.setOnClickListener {
            RxBus.getDefault()
                .post(
                    EventContainer(
                        EventContainer.TYPE_GAME,
                        GameEvent(GameEvent.TYPE_PUZZLE)
                    )
                )
        }
    }

    override fun doAction() {
    }
}
